### ISSUE TITLE
TestGPUTinyTrainingConvergence: tiny PatchTST GPU training run fails to converge

### ISSUE THREAD

**Issue by dndungu** [zerfoo/zerfoo#983]
## Summary

`timeseries.TestGPUTinyTrainingConvergence` fails: loss increases instead of decreasing across a tiny GPU training run.

Found while re-verifying GPU test packages on real hardware (a prior session had wrongly excluded these as "no GPU in this sandbox" -- see zerfoo devlog 2026-08-10). Confirmed **pre-existing**, not caused by the GQA fix in #980 -- reproduces identically with that change reverted (loss values differ slightly between the two runs below since the test appears to use non-fixed randomness, but both runs fail the same convergence assertion in the same direction).

## Repro

```
go test -v -run TestGPUTinyTrainingConvergence ./timeseries/...
```

## Failure output

With the GQA fix (fusedRepeatInterleaveEnabled=false) applied:
```
patchtst_gpu_convergence_test.go:100: GPU tiny train: loss[0]=0.908561 loss[1]=1.222244 elapsed=215.741878ms
patchtst_gpu_convergence_test.go:111: GPU training did not converge: loss[1]=1.222244 not < loss[0]*0.9=0.817705 (loss[0]=0.908561)
--- FAIL: TestGPUTinyTrainingConvergence
```

With the GQA fix reverted (fused RepeatInterleave path active, same as pre-#980 main):
```
patchtst_gpu_convergence_test.go:100: GPU tiny train: loss[0]=1.087949 loss[1]=1.649036 elapsed=203.391162ms
patchtst_gpu_convergence_test.go:111: GPU training did not converge: loss[1]=1.649036 not < loss[0]*0.9=0.979154 (loss[0]=1.087949)
--- FAIL: TestGPUTinyTrainingConvergence
```

Both runs: loss goes UP between the two measured points instead of down, failing the test's `loss[1] < loss[0]*0.9` convergence assertion. Not yet root-caused -- PatchTST (timeseries architecture under test) does not appear to use `GroupedQueryAttention`, consistent with this being unrelated to the GQA/RepeatInterleave bug in #980.

## Environment

- NVIDIA GB10, aarch64, CUDA 13.0, driver 580.159.03

## Note

Not yet investigated beyond confirming it reproduces and is unrelated to the concurrent GQA work. Worth checking learning rate / step count sanity for the "tiny" config, whether this is a flaky small-sample-size convergence assertion versus a real gradient/optimizer bug, and whether it's specific to the GPU path (i.e. does a CPU equivalent converge with the same config?).
